llama_index/llms/generic_utils.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, Callable, Sequence
+
+from llama_index.llms.base import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseGen,
+)
+
+
+def messages_to_prompt(messages: Sequence[ChatMessage]) -> str:
+    """Convert messages to a string prompt."""
+    string_messages = []
+    for message in messages:
+        role = message.role
+        content = message.content
+        string_message = f"{role}: {content}"
+
+        addtional_kwargs = message.additional_kwargs
+        if addtional_kwargs:
+            string_message += f"\n{addtional_kwargs}"
+        string_messages.append(string_message)
+
+    string_messages.append("assistant: ")
```

Comment:
should we maybe pull these roles into an enum?

Reply:
revised.